Mlem/App/Views/Root/Tabs/Settings/ZoomDraggerSettingsView.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  ZoomDraggerSettingsView.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2025-02-02.
+//
+
+import SwiftUI
+
+enum AnimationPhase: CaseIterable {
+    case slideUp, slideDown, hide, show
+    
+    var circleOffset: CGFloat {
+        switch self {
+        case .slideUp: -50
+        case .slideDown: 50
+        case .hide: 50
+        case .show: 50
+        }
+    }
+    
+    var circleOpacity: CGFloat {
+        switch self {
+        case .slideUp: 1
+        case .slideDown: 1
+        case .hide: 0
+        case .show: 1
+        }
+    }
+    
+    var imageScale: CGFloat {
+        switch self {
+        case .slideUp: 2.0
+        case .slideDown: 0.8
+        case .hide: 0.8
+        case .show: 0.8
+        }
+    }
+    
+    var imageSize: CGFloat {
+        switch self {
+        case .slideUp: 400
+        case .slideDown: 150
+        case .hide: 150
+        case .show: 150
+        }
+    }
+}
+
+struct ZoomDraggerSettingsView: View {
+    @Setting(\.zoomDraggerLocation) var zoomDraggerLocation
+    
+    var body: some View {
+        Form {
+            SettingsHeaderView(
+                title: "Drag-To-Zoom",
+                description: "Zoom the image viewer by dragging on the selected side.") {
+                    ZoomDraggerAnimation()
+                }
+            
+            Picker("Location", selection: $zoomDraggerLocation) {
+                ForEach(ZoomDraggerLocation.allCases, id: \.self) { location in
+                    Label(String(localized: location.label), systemImage: location.systemImage)
+                        .tag(location)
+                }
+            }
+            .labelsHidden()
+            .pickerStyle(.inline)
+        }
+        .labelStyle(.conditional)
```

Comment:
Anywhere we use `SettingsHeaderView` we need to add this `.contentMargins` line to make the padding above the section a little smaller. Annoyingly this can't be put inside of the `SettingsHeaderView` itself (I couldn't find a way to make it work, anyway).

```suggestion
        .labelStyle(.conditional)
        .contentMargins(.top, 16)
```